tools/report-converter/tests/unit/test_ubsan_parser.py

```diff
@@ -14,120 +14,83 @@
-            13, 10,
-            "load of value 7, which is not a valid value for type 'enum E'",
-            "UndefinedBehaviorSanitizer"),
-        Message(
-            os.path.abspath('files/ubsan2.cpp'),
-            21, 7,
-            "load of value 2, which is not a valid value for type 'bool'",
-            "UndefinedBehaviorSanitizer"),
-    ]
-
-    UBSANOutputParserTestCase.ubsan1_repr = ubsan1_repr
-    UBSANOutputParserTestCase.ubsan2_repr = ubsan2_repr
-    UBSANPListConverterTestCase.ubsan1_repr = ubsan1_repr
-    UBSANPListConverterTestCase.ubsan2_repr = ubsan2_repr
-
 
 def teardown_module():
-    """Restore environment after tests have ran."""
+    """ Restore environment after tests have ran. """
     global OLD_PWD
     os.chdir(OLD_PWD)
 
 
 class UBSANPListConverterTestCase(unittest.TestCase):
-    """
-    Test the output of the PListConverter, which converts Messages to plist
-    format.
-    """
+    """ Test the output of the UBSANAnalyzerResult. """
 
     def setUp(self):
-        """Setup the PListConverter."""
-        self.plist_conv = UBSANPlistConverter()
+        """ Setup the test. """
+        self.analyzer_result = UBSANAnalyzerResult()
+        self.cc_result_dir = tempfile.mkdtemp()
 
-    def test_empty(self):
-        """Test for empty Messages."""
-        orig_plist = copy.deepcopy(self.plist_conv.plist)
+    def tearDown(self):
+        """ Clean temporary directory. """
+        shutil.rmtree(self.cc_result_dir)
 
-        self.plist_conv.add_messages([])
-        self.assertDictEqual(orig_plist, self.plist_conv.plist)
+    def __check_analyzer_result(self, analyzer_result, analyzer_result_plist,
+                                source_files, expected_plist):
+        """ Check the result of the analyzer transformation. """
 
-        output = StringIO()
-        self.plist_conv.write(output)
+        self.analyzer_result.transform(analyzer_result, self.cc_result_dir)
 
-        with open('empty.plist') as pfile:
-            exp = pfile.read()
-            self.assertEqual(exp, output.getvalue())
+        plist_file = os.path.join(self.cc_result_dir, analyzer_result_plist)
+        with open(plist_file) as pfile:
+            res = plistlib.readPlist(pfile)
 
-        output.close()
+            # Use relative path for this test.
+            res['files'] = source_files
 
-    def test_ubsan1(self):
-        """Test for the ubsan1.plist file."""
-        self.plist_conv.add_messages(self.ubsan1_repr)
+        with open(expected_plist) as pfile:
+            exp = plistlib.readPlist(pfile)
 
-        # use relative path for this test
-        self.plist_conv.plist['files'][0] = 'files/ubsan1.cpp'
+        self.assertEqual(res, exp)
 
-        output = StringIO()
-        self.plist_conv.write(output)
+    def test_empty1(self):
+        """ Test for empty Messages. """
+        with self.assertRaises(SystemExit):
+            self.analyzer_result.transform('empty1.out', self.cc_result_dir)
```

Comment:
I'm not sure transform should exit, returning some error/exception would be better. With an error you have more flexibility and you can still exit with an error code in the main function if that is what should be done.